### PR TITLE
optimize: RM, TM startup connect server fail fast

### PR DESCRIPTION
**Pull request by jsbxyyx**
- [ ] I have registered the PR [changes](../changes).

### Ⅱ. Does this pull request fix one issue?

fixes: #5484

### CODE REVIEW

File: core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java [hunk 1]
```diff
@@ -181,15 +185,20 @@ void reconnect(String transactionServiceGroup) {
...
                     acquireChannel(serverAddress);
                     channelAddress.add(serverAddress);
                 } catch (Exception e) {
                     LOGGER.error("{} can not connect to {} cause:{}", FrameworkErrorCode.NetConnect.getErrCode(),
                         serverAddress, e.getMessage(), e);
+                    invalidAddress += serverAddress + ",";
                 }
             }
+            if (clientCheckChannelFastFail && StringUtils.isNotBlank(invalidAddress)) {
+                throw new FrameworkException("can not connect to [" + invalidAddress + "]");
```
Comment (funky-eyes) [inline core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java:200]: 这个异常如果抛出去，定时任务那块是不是就会不跑了？

Comment (jsbxyyx) [reply core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java:200]: 是的

File: core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java [hunk 2]
```diff
@@ -181,13 +187,29 @@ void reconnect(String transactionServiceGroup) {
...
+                    LOGGER.error("{} can not connect to {} cause:{}",
+                            FrameworkErrorCode.NetConnect.getErrCode(), failedMap.keySet(),
+                            failedMap.values().stream().map(Throwable::getMessage).collect(Collectors.toSet()));
+                } else if (LOGGER.isDebugEnabled()) {
+                    failedMap.forEach((key, value) -> {
+                        LOGGER.error("{} can not connect to {} cause:{} trace information:{}",
+                                FrameworkErrorCode.NetConnect.getErrCode(), key, value.getMessage(), value);
+                    });
+                }
+                String invalidAddress = StringUtils.join(failedMap.keySet().iterator(), ", ");
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java:210]: Implementation problems.failFast is only used for startup logic and should not be included in public processing logic.

File: core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java [hunk 3]
```diff
@@ -161,8 +164,9 @@ void destroyChannel(String serverAddress, Channel channel) {
      * Reconnect to remote server of current transaction service group.
      *
      * @param transactionServiceGroup transaction service group
+     * @param failFast true: check connect failed throws exception. false: not check connect failed.
      */
-    void reconnect(String transactionServiceGroup) {
+    void reconnect(String transactionServiceGroup, boolean failFast) {
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java:169]: This parameter is not elegant.

Comment (slievrly) [reply core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java:169]: No additional parameters are recommended.

**Review by funky-eyes — APPROVED** [review]
LGTM Please modify the commit target branch to 2.x

**Pull request closed without merging**